packages/filebrowser/src/browser.ts

```diff
@@ -170,6 +170,22 @@ export class FileBrowser extends SidePanel {
     }
   }
 
+  /**
+   * Whether to sort notebooks above other files
+   */
+  get sortNotebooksFirst(): boolean {
+    return this._sortNotebooksFirst;
+  }
+
+  set sortNotebooksFirst(value: boolean) {
+    if (this.listing.setNotebooksFirstSorting) {
```

Comment:
What is the reason for testing for `setNotebooksFirstSorting`?

As the `listing` is of type `DirListing` (no interface used), it will always exist, isn't it?

Reply:
I couldn't find why the `setColumnVisibility` was being checked. I believe it should always exist but I was a bit confused and went on the safe side. Do you have more info about this?

https://github.com/jupyterlab/jupyterlab/blob/d47d50bbeb2db80df09489eb7c9588c29a140d22/packages/filebrowser/src/browser.ts#L164C7-L171

Reply:
Given how downstream swaps `listing`, maybe it should become an interface? (this is not a discussion for this PR though). I just wanted to highlight that unfold extension goes [pretty far with the modifications](https://github.com/jupyterlab-contrib/jupyterlab-unfold/blob/6909f91e05d5bf378f076438c71264840d5e47d9/src/unfold.ts ) and depending on the version of `DirListing` it would load the guard may actually be useful (well at least if this lands in 4.x rather than 4.0). We also wanted to modify listing in jupyterlab-git but can probably make dedicated APIs for that use case.

Reply:
Probably this condition was indeed added to make it easier for downstream applications to customize the file browser.

Thanks for providing the extra context, agree we can track follow-ups in other issues :+1:

Reply:
Ok then I'll agree we should keep the safe guard test. But I still would like to switch to a getter/setter pattern. @tpatel, you could replace the safeguard condition by

```ts
if (this.notebooksFirstSorted !== undefined)
```

Reply:
> switch to a getter/setter pattern

Would it be possible to address this in a separate PR? And address this for `setColumnVisibility` as well for consistency?

Reply:
I think it makes sense to address moving both methods to the getter/setter pattern in a separate PR to avoid being in a state with two different patterns 👍